### PR TITLE
backend/hotfix/deadlock-fix

### PR DESCRIPTION
**Pull request by piyushKumar-1**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
1. Fixed deadlock case in sending ride requests (by removing transaction blocks).
2. Optimised driver location update query.
3. Optimised driver availability read query.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by srid — CHANGES_REQUESTED** [review]
`flake.lock` should only have shared-kernel relates changes.

File: flake.lock [hunk 1]
```diff
@@ -1,5 +1,26 @@
 {
   "nodes": {
+    "arion": {
```
Comment (srid) [inline flake.lock:3]: Why is this changed?

File: flake.lock [hunk 2]
```diff
@@ -84,12 +110,16 @@
       }
     },
     "cachix-push": {
+      "inputs": {
+        "cachix": "cachix",
+        "devour-flake": "devour-flake"
```
Comment (srid) [inline flake.lock:115]: Why is this changed?

File: flake.lock [hunk 3]
```diff
@@ -321,6 +354,32 @@
         "type": "github"
       }
     },
+    "devenv_2": {
+      "inputs": {
+        "flake-compat": [
+          "common",
```
Comment (srid) [inline flake.lock:360]: Also this?

So many unnecessary changes.

You should revert the whole `flake.lock` change, and then update it correctly using `nix flake lock`.

Comment (piyushKumar-1) [reply flake.lock:360]: sure, this PR is for syncing the last prod release though, not for the redis change.

**Review by srid — APPROVED** [review]
